warehouse/integrations/middleware/sqlquerywrapper/sql.go

```diff
@@ -137,78 +205,101 @@ func (db *DB) WithTx(ctx context.Context, fn func(*Tx) error) error {
 	return tx.Commit()
 }
 
-func (db *DB) logQuery(query string, elapsed time.Duration) {
-	if elapsed < db.slowQueryThreshold {
-		return
-	}
+func (db *DB) logQuery(query string, since time.Time) logQ {
+	return func() {
+		if db.slowQueryThreshold <= 0 {
+			return
+		}
+		if db.since(since) < db.slowQueryThreshold {
+			return
+		}
+
+		sanitizedQuery, _ := misc.ReplaceMultiRegex(query, db.secretsRegex)
 
-	sanitizedQuery, _ := misc.ReplaceMultiRegex(query, db.secretsRegex)
+		keysAndValues := []any{
+			logfield.Query, sanitizedQuery,
+			logfield.QueryExecutionTime, db.since(since),
+		}
+		keysAndValues = append(keysAndValues, db.keysAndValues...)
 
-	keysAndValues := []any{
-		logfield.Query, sanitizedQuery,
-		logfield.QueryExecutionTime, elapsed,
+		db.logger.Infow("executing query", keysAndValues...)
 	}
-	keysAndValues = append(keysAndValues, db.keysAndValues...)
+}
 
-	db.logger.Infow("executing query", keysAndValues...)
+type logQ func()
+
+func (tx *Tx) GetTx() *sql.Tx {
+	return tx.Tx
 }
 
+// Begin starts a transaction.
+//
+// Use BeginTx to pass context and options to the underlying driver.
 func (db *DB) Begin() (*Tx, error) {
-	if tx, err := db.DB.Begin(); err != nil {
-		return nil, err
-	} else {
-		return &Tx{tx, db}, nil
-	}
+	return db.BeginTx(context.Background(), nil)
 }
 
 func (db *DB) BeginTx(ctx context.Context, opts *sql.TxOptions) (*Tx, error) {
-	if tx, err := db.DB.BeginTx(ctx, opts); err != nil {
+	ctx, cancel := queryContextWithTimeout(ctx, db.transactionTimeout)
+	defer cancel()
```

Comment:
wonder how the tests passed considering the following documentation for `*sql.DB.BeginTx`:

```
func (*sql.DB).BeginTx(ctx context.Context, opts *sql.TxOptions) (*sql.Tx, error)
BeginTx starts a transaction.

The provided context is used until the transaction is committed or rolled back. 
If the context is canceled, the sql package will roll back the transaction. 
Tx.Commit will return an error if the context provided to BeginTx is canceled.

The provided TxOptions is optional and may be nil if defaults should be used. 
If a non-default isolation level is used that the driver doesn't support, an error will be returned.
```

Reply:
made changes so context is cancelled when commit/rollback returns.

Reply:
> wonder how the tests passed

We weren't setting `db.transactionTimeout`.
so `defer cancel()` wouldn't cancel the context.
We'd have seen errors if we'd set that timeout.

Adding tests to verify that.